src/engine/BMInterfaceTournament.php

```diff
@@ -0,0 +1,1108 @@
+        return TRUE;
+    }
+
+    /**
+     * Validate whether a player can cancel a tournament
+     *
+     * @param int $userId
+     * @param int $tournamentId
+     * @param BMTournament $tournament
+     * @return bool
+     */
+    protected function validate_cancel_tournament($userId, $tournamentId, $tournament) {
+        if (is_null($tournament)) {
+            $this->set_message("Tournament $tournamentId does not exist");
+            return NULL;
+        }
+
+        if ($tournament->tournamentState > BMTournamentState::JOIN_TOURNAMENT) {
+            $this->set_message('The tournament has already started.');
+            return NULL;
+        }
+
+        if ($tournament->creatorId !== $userId) {
+            $this->set_message('You did not create this tournament.');
+            return NULL;
+        }
+
+        return TRUE;
+    }
+
+    /**
+     * Attempt to cancel a tournament in the database
+     *
+     * @param int $userId
+     * @param int $tournamentId
+     * @return bool
+     */
+    protected function cancel_tournament_in_database($userId, $tournamentId) {
+        // query is written in this way to avoid possible race conflicts between
+        // multiple people trying to act on a tournament at the same time
+        //
+        // a tournament can only be deleted by the creator if it has not started yet
+        $query = 'UPDATE tournament '.
+                 'SET status_id = (SELECT id FROM tournament_status WHERE name = "CANCELLED"), '.
+                 'tournament_state = :tournament_state '.
+                 'WHERE id = :tournament_id '.
+                 'AND tournament_state <= 20 '.
+                 'AND creator_id = :creator_id';
+        $parameters = array(':tournament_id' => $tournamentId,
+                            ':tournament_state' => BMTournamentState::CANCELLED,
+                            ':creator_id' => $userId);
+
+        return self::$db->update_and_report_if_changed($query, $parameters);
+    }
+
+    /**
+     * Dismiss tournament link from overview page
+     *
+     * @param int $playerId
+     * @param int $tournamentId
+     * @return bool
+     */
+    public function dismiss_tournament($playerId, $tournamentId) {
+        try {
+            $query1 =
+                'SELECT s.name AS "status" ' .
+                'FROM tournament AS t ' .
+                'INNER JOIN tournament_status AS s ON s.id = t.status_id ' .
+                    'LEFT JOIN tournament_player_map AS m ' .
+                    'ON m.tournament_id = t.id AND m.player_id = :player_id ' .
+                'WHERE t.id = :tournament_id';
+            $parameters = array(
+                ':player_id' => $playerId,
+                ':tournament_id' => $tournamentId,
+            );
+            $columnReturnTypes = array(
+                'status' => 'str',
+            );
+
+            $row = self::$db->select_single_value($query1, $parameters, $columnReturnTypes);
```

Comment:
So, can you fix this without a test?  It's blowing up too much for me to get a valid response, so i could get you a responder test, but it would be time-consuming to construct.  I called this out in https://github.com/buttonmen-dev/buttonmen/pull/2131#issuecomment-1407554494

Basically this is simply not the syntax of `BMDB->select_single_value()` --- it doesn't take an array as its last parameter, it takes a string.  You can see this every other place you call this function.

If you want to fix this with the least possible typing, just do:
```diff
- $row = self::$db->select_single_value($query1, $parameters, $columnReturnTypes);
+ $row = self::$db->select_rows($query1, $parameters, $columnReturnTypes);
```

You could make the syntax more concise, but you'd make my life easier in test if you just changed that single word, and it definitely does not work right now.

Reply:
Taking a look now.

Reply:
Just in passing, I note that line 1092 shows that a dismiss also does an unwatch, so that may be relevant for the last point we were discussing.